From a1ed5f4f104df70788849cd274544ad88c27ffe6 Mon Sep 17 00:00:00 2001
From: Wenlong Zhuang <daisen.zhuang@rock-chips.com>
Date: Tue, 25 Dec 2018 17:24:36 +0800
Subject: [PATCH] arm64: dts: rockchip: rk1808-evb-x4: enable virtual camera,
 cif and mipi-dphy-rx

Init resolution 3840x720 and bus format MEDIA_BUS_FMT_SBGGR8_1X8
for virtual camera.

Change-Id: Id45f7deb3896ca230cfd3fc44238c0d62d79416a
Signed-off-by: Wenlong Zhuang <daisen.zhuang@rock-chips.com>
---
 .../arm64/boot/dts/rockchip/rk1808-evb-x4.dts | 66 +++++++++++++++++++
 1 file changed, 66 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts b/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts
index 0bc2b12b451a..3abbe64cbda7 100644
--- a/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts
+++ b/arch/arm64/boot/dts/rockchip/rk1808-evb-x4.dts
@@ -29,6 +29,21 @@
 /delete-node/ &vdd1v5_dvp;
 /delete-node/ &vcc2v8_dvp;
 
+&cif {
+	status = "okay";
+
+	port {
+		cif_in: endpoint@0 {
+			remote-endpoint = <&dphy_rx_out>;
+			data-lanes = <1 2 3 4>;
+		};
+	};
+};
+
+&cif_mmu {
+	status = "okay";
+};
+
 &emmc {
 	status = "disabled";
 };
@@ -37,6 +52,25 @@
 	status = "disabled";
 };
 
+&i2c0 {
+	status = "okay";
+
+	vcamera@30 {
+		compatible = "rockchip,virtual-camera";
+		reg = <0x30>;
+		width = <3840>;
+		height = <720>;
+		bus-format = <MEDIA_BUS_FMT_SBGGR8_1X8>;
+
+		port {
+			vcamera_out: endpoint {
+				remote-endpoint = <&dphy_rx_in>;
+				link-frequencies = /bits/ 64 <500000000>;
+			};
+		};
+	};
+};
+
 &i2c1 {
 	status = "disabled";
 };
@@ -49,6 +83,38 @@
 	status = "okay";
 };
 
+&mipi_dphy_rx {
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@0 {
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_rx_in: endpoint@1 {
+				reg = <1>;
+				remote-endpoint = <&vcamera_out>;
+				data-lanes = <1 2 3 4>;
+			};
+		};
+
+		port@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			dphy_rx_out: endpoint@0 {
+				reg = <0>;
+				remote-endpoint = <&cif_in>;
+			};
+		};
+	};
+};
+
 &rk809_codec {
 	status = "disabled";
 };
-- 
2.35.3

